Add option to list available stable versions and URL

cixtor opened the pull request:
Option `-available` will send a HTTP GET request to the download page of the official website of the Go project, then it will find all the HTML tags identified by a CSS class name _"toggle"_ or _"toggleVisible"_ for the latest stable version. Then it will process the text and print the list in the console. Note that I am also modifying the content of the Git ignore file to allow the addition of the test files, it seems that the current exclusion rules are targeting everything inside the parent source directory instead of the unnecessary files inside the child source directory located inside `src/gobu/`

**EDIT:** I have added the timeout, by default it will be 5 secs, I was thinking to move that number into an individual variable _(maybe a global)_ or an environment variable so people can override it by hand, but then thought that it was premature to think about that now. Regarding the test cases, I don't have much experience mocking HTTP requests, I decided to separate the code that sends the HTTP request from the code that parses and extracts the relevant data, then adding a new parameter in the all new functions associated with this feature to expect an _io.Reader_ which in the tests is built by hand. Let me know if there is a better way to do this, as of now the execution of the tests will not send real HTTP requests so that is something, but if there is a better way to mock this I am happy to learn.

cixtor committed:
Add option to list available stable versions and URL

Option -available will send a HTTP GET request to the download page of
the official website of the Go project, then it will find all the HTML
tags identified by a CSS class name "toggle" or "toggleVisible" for the
latest stable version. Then it will process the text and print the list
in the console. Note that I am also modifying the content of the Git
ignore file to allow the addition of the test files, it seems that the
current exclusion rules are targetting everything inside the parent
source directory instead of the unnecessary files inside the child
source directory located inside src/gobu/

dz0ny commented on src/gobu/remote.go:
```diff
@@ -31,3 +35,65 @@ func download(version string) {
 }
+
+func availableVersions() []string {
+	client := &http.Client{}
```
Add timeout, by default it's ~inf so it migh block.

dz0ny commented on src/gobu/remote_test.go:
```diff
@@ -0,0 +1,33 @@
+import "testing"
+import "regexp"
+
+func CheckVersionName(t *testing.T, name string) {
```
Also mock http client, so that we don't query actual server on each test.

dz0ny reviewed (changes requested):
Looks good, some minor changes needed thou.

cixtor committed:
Add timeout for the option to list available versions

cixtor committed:
Modify available versions command to expect reader

To facilitate the testing of the functions used to extract the list of
stable versions from the download page of the official Go website this
commit will separate the HTTP request code from the code that parses and
extracts the relevant information from the HTML code. This way we can
pass a different buffer in the test files with static data and avoid
unnecessary requests to the live website in the continuous integration
service.

cixtor committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	src/gobu/remote.go
